### PR TITLE
Update opentelemetry to 0.20.0

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

### CODE REVIEW

**Comment by rubenvp8510** [comment]
@jpkrohling Question, why some tests are using opentracing? should we migrate it to otel?

File: pkg/cmd/start/bootstrap.go [hunk 1]
```diff
@@ -229,39 +229,50 @@ func buildIdentity(ctx context.Context, podNamespace string) {
...
+
 	if err == nil {
-		tracing.AddJaegerExporter(jaegerExporter)
+		traceProvider := tracesdk.NewTracerProvider(
+			tracesdk.WithBatcher(exporter),
+			tracesdk.WithResource(resource.NewWithAttributes(
+				semconv.ServiceNameKey.String("jaeger-operator"),
+				semconv.ServiceVersionKey.String(version.Get().Operator),
+				semconv.ServiceNamespaceKey.String(namespace),
+				semconv.ServiceInstanceIDKey.String(viper.GetString(v1.ConfigIdentity)),
```
Comment (jpkrohling) [inline pkg/cmd/start/bootstrap.go:264]: \o/

File: pkg/controller/jaeger/account.go [hunk 2]
```diff
@@ -3,8 +3,9 @@ package jaeger
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
+
```
Comment (jpkrohling) [inline pkg/controller/jaeger/account.go:7]: Import order :-)

File: pkg/controller/jaeger/clusterrolebinding.go [hunk 3]
```diff
@@ -3,9 +3,10 @@ package jaeger
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
```
Comment (jpkrohling) [inline pkg/controller/jaeger/clusterrolebinding.go:6]: Import order

File: pkg/controller/jaeger/configmap.go [hunk 4]
```diff
@@ -3,8 +3,9 @@ package jaeger
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
+
```
Comment (jpkrohling) [inline pkg/controller/jaeger/configmap.go:7]: And here as well

File: pkg/controller/jaeger/consolelink.go [hunk 5]
```diff
@@ -3,10 +3,11 @@ package jaeger
 import (
 	"context"
 
+	"go.opentelemetry.io/otel"
```
Comment (jpkrohling) [inline pkg/controller/jaeger/consolelink.go:6]: And here

File: pkg/tracing/bootstrap.go [hunk 6]
```diff
@@ -1,40 +1,31 @@
...
-		log.Fatal(err)
-	}
-
-	global.SetTraceProvider(provider)
+	provider := sdktrace.NewTracerProvider(sdktrace.WithSampler(sdktrace.AlwaysSample()))
+	otel.SetTracerProvider(provider)
 }
 
 // AddJaegerExporter includes the given exporter into the existing provider
-func AddJaegerExporter(exporter *jaeger.Exporter) {
```
Comment (jpkrohling) [inline pkg/tracing/bootstrap.go:30]: Just remove this code entirely :-)

File: pkg/controller/jaeger/dependencies.go [hunk 7]
```diff
@@ -44,19 +45,18 @@ func (r *ReconcileJaeger) handleDependencies(ctx context.Context, str strategy.S
...
-		key.String("dependency.namespace", dep.Namespace),
+		otelattribute.String("dependency.name", dep.Name),
+		otelattribute.String("dependency.namespace", dep.Namespace),
 	)
 
 	err := r.client.Create(ctx, &dep)
 	if err != nil && !apierrors.IsAlreadyExists(err) {
-		span.SetStatus(codes.Internal)
-		span.SetAttribute(key.String("error", err.Error()))
+		span.SetStatus(codes.Error, err.Error())
```
Comment (jpkrohling) [inline pkg/controller/jaeger/dependencies.go:59]: Looks like `tracing.HandleError` is already taking care of the `span.SetStatus`? You can probably just remove this line (and the same for the other places).

**Review by jpkrohling — COMMENTED** [review]
Looks great! There are problems with the import order, and a small comment about error handling, but other than that, looks good to me.

**Comment by jpkrohling** [comment]
One more comment: looks like there are some spans missing. There used to be a "bootstrap'' operation, which isn't being shown now after this update:

image

This is what we had before:

image

**Comment by rubenvp8510** [comment]
I did some modifications to bootstraping process.
This is how it looks now:

Screenshot from 2021-05-15 17-52-56

**Comment by jpkrohling** [comment]
The following test failed and I just restarted: 

```
        --- FAIL: kuttl/harness/allinone-ingress (45.83s)
```

**Comment by rubenvp8510** [comment]
@jpkrohling This is ready for another review :)

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**